src/installer.ts

```diff
@@ -132,7 +132,13 @@ async function installGoVersion(
   auth: string | undefined
 ): Promise<string> {
   core.info(`Acquiring ${info.resolvedVersion} from ${info.downloadUrl}`);
-  const downloadPath = await tc.downloadTool(info.downloadUrl, undefined, auth);
+
+  // Windows requires that we keep the extension (.zip) for extraction
+  const isWindows = os.platform() === 'win32';
+  const tempDir = process.env.RUNNER_TEMP || '.';
```

Comment:
I am not sure that we need to add tempDir. As I know `downloadTool` command should handle it.

Reply:
IIRC, it only does it if dest is undefined. In the Windows case we provide a file name which ends up relative. This variable is only used in the Windows case, on others we let downloadTool handle all of it.

Reply:
I've checked it. You're right. In this case I think it should work. Thank you.